components/shared/Select.tsx

```diff
@@ -0,0 +1,76 @@
+import { Fragment, useState } from 'react'
+import { Listbox, Transition } from '@headlessui/react'
+import { CheckIcon, ChevronUpDownIcon } from '@heroicons/react/20/solid'
+import { FC } from 'react';
+
+interface Props {
+    options:{ id: number; name: string; }[]
+}
+  
+  function classNames(...classes: any) {
+    return classes.filter(Boolean).join(' ')
+  }
+
+export const Select: FC<Props> = ({ options }) => {
+
+    const [selected, setSelected] = useState(options[0])
```

Comment:
I assume this is a work in progress but I don't see a way to communicate with the parent on which value was selected.